src/Microsoft.CodeAnalysis.BannedApiAnalyzers/Core/SymbolIsBannedAnalyzerBase.cs

```diff
@@ -156,19 +150,62 @@ private void OnCompilationStart(CompilationStartAnalysisContext compilationConte
 
             return;
 
+            bool IsBannedSymbol([NotNullWhen(true)] ISymbol? symbol, [NotNullWhen(true)] out BanFileEntry? entry)
+            {
+                if (symbol is { ContainingSymbol.Name: string parentName } &&
+                    bannedApis.TryGetValue((parentName, symbol.Name), out var entries))
+                {
+                    foreach (var bannedFileEntry in entries)
+                    {
+                        foreach (var bannedSymbol in bannedFileEntry.Symbols)
+                        {
+                            if (SymbolEqualityComparer.Default.Equals(symbol, bannedSymbol))
+                            {
+                                entry = bannedFileEntry;
+                                return true;
+                            }
+                        }
+                    }
+                }
+
+                entry = null;
+                return false;
+            }
+
+            bool ShouldAnalyzeAttributes()
+            {
+                foreach (var kvp in bannedApis)
+                {
+                    if (!kvp.Key.SymbolName.EndsWith("Attribute", StringComparison.InvariantCulture))
```

Comment:
>if (!kvp.Key.SymbolName.EndsWith("Attribute", StringComparison.InvariantCulture)) [](http://example.com/codeflow?start=20&length=81)

This is a heuristic that should be true in most cases, right? If so, might be nice to indicate so and why we use it to prevent getting the symbols.

Reply:
sure.  will doc.